<!-- Starts up all nodes for running the robot -->
<launch>
    <!-- Prints output messages to screen for all nodes -->
    <arg name="debug" default="true" />

    <!-- set names of sensors -->
    <arg name="camera1" default="camera1" />
    <arg name="camera2" default="camera2" />


    <!-- Start generating transforms -->
    <include file="$(find setup_tf)/launch/setup_tf.launch">
        <arg name="debug" value="$(arg debug)" />
    </include>

    <!-- Load Camera Calibrations -->
    <group ns="camera1/calibration_rgb">
        <rosparam file="$(find cameras)/calibrations/camera1_rgb.yaml" command="load" />
    </group>

    <group ns="camera2/calibration_rgb">
        <rosparam file="$(find cameras)/calibrations/camera2_rgb.yaml" command="load" />
    </group>

    <!-- Load markers -->
    <rosparam file="$(find cameras)/markers/marker_list.yaml" command="load" />

    <!-- Camera Node -->
    <include file="$(find astra_launch)/launch/astra.launch">
        <arg name="camera" value="$(arg camera1)" />
    </include>
     
    <include file="$(find usb_cam)/launch/astrapro_rgb.launch">
        <arg name="camera" value="$(arg camera2)" />
    </include>
   


    <node pkg="cameras" name="camera_posev1" type="camera_posev_last" respawn="true" output="screen">
        <!--remap from="/camera/rgb/image_raw" to="/$(arg camera1)/rgb/image_raw" -->
        <param name="camera" value="$(arg camera1)" />
        <param name="display_image" value="true" />
    </node>

    
    <node pkg="cameras" name="camera_posev2" type="camera_posev_last" respawn="true" output="screen">
   
        <!--remap from="/camera/rgb/image_raw" to="/$(arg camera2)/rgb/image_raw" -->
        
        <param name="camera" value="$(arg camera2)" />
        <param name="display_image" value="true" />
    </node>
    




</launch>
